pkg/smclient/client.go

```diff
@@ -51,9 +53,43 @@ type serviceManagerClient struct {
 	httpClient auth.Client
 }
 
-// NewClient returns new SM client
-func NewClient(httpClient auth.Client, config *ClientConfig) Client {
-	return &serviceManagerClient{config: config, httpClient: httpClient}
+// NewClientWithAuth returns new SM Client configured with the provided configuration
+func NewClientWithAuth(httpClient auth.Client, config *ClientConfig) (Client, error) {
+	if httpClient == nil {
```

Comment:
```suggestion
	if httpClient == nil {
	    httpClient = defaultHTTPClient
	}
```
This way it can be tested

Reply:
It's the same, the problem is that I need a mock for OAuth server that generates tokens. The one from SM core cannot be used as it generates some strange Ginkgo errors in the tests